public/store/context/react_flow_context_provider.tsx

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { createContext, useState } from 'react';
+
+const initialValues = {
+  reactFlowInstance: null,
+  setReactFlowInstance: () => {},
+  deleteNode: (nodeId: string) => {},
+  deleteEdge: (edgeId: string) => {},
+};
+
+export const rfContext = createContext(initialValues);
+
+/**
+ * This returns a provider from the rfContext context created above. The initial
+ * values are set so any nested components can use useContext to access these
+ * values.
+ *
+ * This is how we can manage ReactFlow context consistently across the various
+ * nested child components.
+ */
+export function ReactFlowContextProvider({ children }: any) {
+  const [reactFlowInstance, setReactFlowInstance] = useState(null);
+
+  const deleteNode = (nodeId: string) => {
```

Comment:
Why isn't this where the `onConnect` `onDrop` and other workspace functions are. I understand why it can live here also but surprised its not all in the same place as other functions

Reply:
It is part of the context so the functions will be easily accessible by other React components later on that are not directly stored in the main `Workspace` component (and may not even be children of `Workspace`, such as helper menus to delete the component, or handlers for removing connections of edges of a UI component, etc.

The `onConnect`/`onDrop`/etc. fns are included just in the `Workspace` for now, but may potentially be refactored or added into the context later on if that pattern will make sense. Right now, the configuration I have matches with the Flowise pattern which is beneficial for us to start with for now and change as needed down the line.